### ISSUE TITLE
[ChC] [ThaumCraft] Rainbow Forest seems to be enlarging Aura Nodes several times

### ISSUE THREAD

**Issue by Shahelion** [ReikaKalseki/Reika_Mods_Issues#2120]
I initially thought it was happening when I interacted with the nodes, but it seems to be happening on its own, without me being nearby sometimes. The three screenshots are of the same node. These were taken over the course of an hour and a half real-time.

![2019-06-02_11 22 32](https://user-images.githubusercontent.com/6917535/58766564-2f9ba080-8535-11e9-9954-ce3668c94815.png)
![2019-06-02_11 42 33](https://user-images.githubusercontent.com/6917535/58766565-2f9ba080-8535-11e9-92f2-b8dcb2b80c9b.png)
![2019-06-02_12 45 07](https://user-images.githubusercontent.com/6917535/58766566-2f9ba080-8535-11e9-8f20-41d97d3db869.png)

**Comment by ReikaKalseki** [comment]
I need more information than this.

**Comment by Shahelion** [comment]
What other information do you need? I'll see what I can do

Mod list:
https://pastebin.com/pPe79zHP

**Comment by ReikaKalseki** [comment]
No, situational information.

**Comment by Shahelion** [comment]
I'm not sure what else specifically you need. I haven't been able to pinpoint yet what is causing this. It's happened while actively draining a node, while just looking at it from a fair distance away, and even while the chunk isn't even visible/loaded.

**Comment by ReikaKalseki** [comment]
Has it affected nodes with which you have never interacted?

**Comment by Shahelion** [comment]
Hmm...Not sure, since I've probably interacted with most but not all of the nodes in this area. Will report back after some testing.

**Comment by lAimBotl** [comment]
Have you tested nodes not in a rainbow forest?

**Comment by Shahelion** [comment]
So far, every affected node has been in a rainbow forest biome. I have several nodes just outside it that are still small but get interacted with just as much (if not more, since I can actually see them)

**Comment by lAimBotl** [comment]
I remember the crc book says that trees of the same color as the pylon make it recharge faster, and the pylon (last I checked) also had aspects in it like a aura node, maybe it is related to that?

**Comment by ReikaKalseki** [comment]
No, that only applies to pylons.

**Comment by ReikaKalseki** [comment]
I need someone to try to generate some rainbow forest nodes, and then inspect their values, then repeatedly load and unload the area and see if anything changes.

**Comment by ReikaKalseki** [comment]
So...

**Comment by ReactivePlatypus** [comment]
I've done a little testing of this in my own modpack (very different modlist, using V23b of Chromaticraft and DragonAPI)) and can confirm that this does happen and affects both player-spawned and naturally generated nodes, regardless of whether the player has scanned or interacted with them. Here's a naturally-generated node when it was first generated. Note that it is already very large for a regular aura node.
![2019-10-31_12 27 52](https://user-images.githubusercontent.com/22828039/67973799-32cbb300-fbdf-11e9-9345-173798eee35d.png)
(It's hard to read the thaumometer text - the aspect values are 207 Aer and 134 Aqua)
And this was how it looked after repeatedly teleporting to and from the area.
![2019-10-31_12 32 50](https://user-images.githubusercontent.com/22828039/67972096-02cee080-fbdc-11e9-8f1d-875a94a11cb0.png)
Additionally, this was a node spawned by me nearby.
![2019-10-31_12 31 26](https://user-images.githubusercontent.com/22828039/67972159-1e39eb80-fbdc-11e9-9cc8-cee56f5c1328.png)
After teleporting to and from the site about 5 times, it looked like this. It showed no obvious change after any of the previous teleports.
![2019-10-31_12 33 45](https://user-images.githubusercontent.com/22828039/67972189-2eea6180-fbdc-11e9-8ea0-36bf7c62f463.png)
Both of these nodes were placed within projection range (NOT line of sight, though) of a magenta crystal pylon. They were also not in bullying range of any other nodes. I initially thought that they must be plugging into the pylon somehow, but then this happened.
![2019-10-31_12 33 59](https://user-images.githubusercontent.com/22828039/67972490-c8b20e80-fbdc-11e9-9739-af2a21de02c4.png)
Unfortunately, I was dumb and didn't grab a screenshot of this node in its original state, but its aspect counts were all below 50 originally, and it was not Bright. This node was 84m away from the pylon, so it couldn't have taken power form it.
![2019-10-31_12 28 59](https://user-images.githubusercontent.com/22828039/67972261-517c7a80-fbdc-11e9-9606-d30ae5fd1a9a.png)
I placed several other aura nodes nearby and none of them gained any aspects after ~50 loads/unloads, though many of them turned bright when they were originally not. I also placed a group of nodes directly outside the rainbow forest and they were completely unchanged.

I did some more testing with groups of nodes in bullying range. An unprotected 4x4 node group simply bullied each other to death and the result was a medium-sized pale node, as normal. I repeated the process using node stabilizers to keep them isolated. After around 30 load/unload cycles, they were all unaffected. That was also the case with advanced stabilizers, which allow nodes to bully but not be bullied. I was about to go ahead and say bullying wasn't the cause, and went to check on the nodes outside the rainbow biome again (~7 chunks away), but when I got back, this happened:
![2019-10-31_13 19 30](https://user-images.githubusercontent.com/22828039/67974901-34967600-fbe1-11e9-8ee8-cb9510514a37.png)
One node in particular grew to *enormous* size, as large as an EMP'ed supernode.
![2019-10-31_13 20 37](https://user-images.githubusercontent.com/22828039/67975032-66a7d800-fbe1-11e9-921f-5f814482bf65.png)
(the unknown aspect in the screenshot was Alienis, for the record)
I don't know if node bullying was the cause of this or not, but at 5 out of those 16 nodes gained between 500 and 5000 aspect points just from me flying outside their render range. I tested another 4x4 unprotected node group in another part of the rainbow forest roughly 200m away from the pylon, and they just bullied each other to death as expected. No absurd node growth after multiple comings and goings by teleport and travel.

tl;dr summary:

 -This does happen both with naturally and artificially placed nodes

-It seems to happen more often with nodes that are close to crystal pylons, but the nodes don't seem to be taking power from them. All affected nodes were within 5 chunks of a pylon and all but one were within 2 chunks. It's independent of whether the node has LOS to the pylon itself.

 -All node growths happened while the nodes were not being watched, and when the player was outside node render distance but not necessarily outside chunk loading range

-It does not happen in the area surrounding the rainbow forest.

-It doesn't seem to affect nodes that have been stabilized

-It seems to happen much more often to nodes that are near other nodes, maybe related to node bullying mechanics?

EDIT: I have now done some testing using an ultra bare-bones pack of just Thaumcraft 4, Baubles, and DragonAPI and Chromaticraft V23b. The bug still happens with all aura nodes within Rainbow Forests. Here's the result of a test with a cluster of bullying nodes similar to in my modpack test:
![2019-10-31_14 19 28](https://user-images.githubusercontent.com/22828039/67979403-2b5dd700-fbea-11e9-8f51-816d87d6b186.png)
That was after around 12 teleports back and forth from 6000 blocks away. This group was within a ~4 chunk radius of a magenta pylon, as with the previous test (naturally spawned - I got lucky)
I then tested to see if individual nodes would also still be affected if they were not in bullying range, and it turns out, they are. This group was neither near a pylon nor close enough to each other to bully:
![2019-10-31_14 23 23](https://user-images.githubusercontent.com/22828039/67979590-97d8d600-fbea-11e9-99b1-f80d5f7181db.png)
The growth is not as dramatic, but it's definitely noticeable. This was after 6 teleports.
![2019-10-31_14 23 25](https://user-images.githubusercontent.com/22828039/67979616-a4f5c500-fbea-11e9-9c58-d79d6a059662.png)
This one in particular tripled all its base aspect counts and gained Potentia aspect seemingly out of nowhere. It wasn't even close to any other nodes containing Ignis.

**Comment by ReikaKalseki** [comment]
I have a feeling this has something to do with my worldgen interception system firing during chunk load instead of just generation...
